force-app/classes/src/Assertions.cls

```diff
@@ -6,76 +6,137 @@
  */
 @IsTest
 public class Assertions {
-  public virtual class SystemAsserter {
-    public virtual void assertEquals(
-      Object expected,
-      Object actual,
-      String message
-    ) {
-      System.assertEquals(expected, actual, message);
+  public interface MethodSpyAssertions {
+    /**
+     * Assert that a method spy has never been called
+     */
+    void hasNotBeenCalled();
+
+    /**
+     * Assert that a method spy has been called at least one time with any parameters
+     */
+    void hasBeenCalled();
+
+    /**
+     * Assert that a method spy has been called exactly N times with any parameters
+     *
+     * @param count number of times the method spy should have been called
+     */
+    void hasBeenCalledTimes(final Integer count);
+
+    /**
+     * Assert that a method spy has been called at least one time with matching parameters
+     *
+     * @param params params that should match one call paramters
+     * @see          Params
+     */
+    void hasBeenCalledWith(final Params params);
+
+    /**
+     * Assert that a method spy has been lastly called with matching parameters
+     *
+     * @param params params that should match last call paramters
+     * @see          Params
+     */
+    void hasBeenLastCalledWith(final Params params);
+  }
+
+  public interface Asserter {
+    void isTrue(Boolean value, String message);
+    void isFalse(Boolean value, String message);
+  }
+
+  /**
+   * Starts to build assertions on a method spy
+   *
+   * @param spy the MethodSpy to assert
+   * @see       MethodSpy
+   */
+  public static MethodSpyAssertions assertThat(MethodSpy spy) {
+    return assertThat(spy, METHOD_SPY_ASSERTER);
+  }
+
+  /**
+   * Starts to build assertions on a method spy using a custom asserter
+   *
+   * @param spy      the MethodSpy to assert
+   * @param asserter the Assert to use during assertions
+   */
+  public static MethodSpyAssertions assertThat(
```

Comment:
**suggestion:** could be private @testVisible ?

I know it is not important because it will be used in test contest anyway
But it is a way to show it is used for our testing and may not be useful in production context.